Allow chaining in IfUnlessModifier

lumeet opened the pull request:
Don't register an offense for an if-end block if it's immediately
followed by a chained call.

Currently, auto-correction produces changes in AST. That's why I call this a bug.

Before:
```ruby
if test
  something
end.inspect
```

After:
```ruby
something if test.inspect
```

bbatsov commented on spec/rubocop/cop/style/if_unless_modifier_spec.rb:
```diff
@@ -242,4 +242,11 @@
     end
   end
+
+  it 'accepts if-end followed by a chained call' do
```
I think you wanted to write two examples and mixed them or something like this.

lumeet commented:
I certainly did mix up some things. :) Basically, we could solve the issue

1. by fixing auto-correction to produce something like `(something if test).inspect`,
2. by skipping auto-correction, or
3. by accepting it.

(1) looks pretty bad, (2) is what I did first and (3) what I then tried to implement... with some remainders from (2). Hopefully it looks better now.

bbatsov commented on CHANGELOG.md:
```diff
@@ -2,6 +2,10 @@
 
 ## master (unreleased)
 
+### Bug Fixes
```
There's another `bug fixes` block for this release.

bbatsov commented:
> by fixing auto-correction to produce something like (something if test).inspect

Btw, I wouldn't say this looks bad, but I'm OK with your approach.

bbatsov commented on lib/rubocop/cop/style/if_unless_modifier.rb:
```diff
@@ -20,10 +20,16 @@ def on_if(node)
           return if modifier_if?(node)
           return if elsif?(node)
           return if if_else?(node)
+          return if chained?(node)
```
I'd probably add a comment explaining why we check about this at all.

lumeet commented:
Commented, fixed changelog and rebased.

> Btw, I wouldn't say this looks bad, but I'm OK with your approach.

I'd personally try to rethink the solution completely, and if I had to, I'd probably be explicit about the `nil` value and use the ternary operator.

bbatsov commented:
> I'd personally try to rethink the solution completely, and if I had to, I'd probably be explicit about the nil value and use the ternary operator.

Yeah, there's definitely room for improvement here.
